frontend/tests/utils/downloadUtils.ts

```diff
@@ -0,0 +1,400 @@
+    )
+  ).toBeTruthy();
+
+  //check if the 3 column contains the gene expression link
+  expect(
+    options.data[2].includes(`${TEST_URL}${ROUTES.WHERE_IS_MY_GENE}`)
+  ).toBeTruthy();
+
+  // Extract the link using a regular expression
+  const linkRegex = /https?:\/\/[^\s]+/;
+  const linkMatch = options.data[2].match(linkRegex);
+  let text: string | null = "";
+  if (options.filterName !== "no-filter") {
+    //this gets the filter displayed on the ui
+    text = await getFilterText(page, options.filterName);
+  }
+
+  // Check if a match was found and log the result
+  if (linkMatch) {
+    const link = linkMatch[0];
+    //verify link is valid
+
+    expect(link).toEqual(url);
+
+    // wait until the new page fully loads
+    await page.waitForLoadState();
+    // expect the header on the new page to be visible
+    expect(page.getByTestId("download-button")).toBeVisible();
+  }
+
+  verifyFilterValues(
+    options.data,
+    options.filterName,
+    text,
+    options.noSelectionText
+  );
+};
+
+const verifyFilterValues = (
+  data: string[],
+  filterValue: string,
+  filterText: string | null,
+  noSelectionText = "No selection"
+) => {
+  let datasetFilterValue = noSelectionText;
+  let diseaseFilterValue = noSelectionText;
+  let ethnicityFilterValue = noSelectionText;
+  let sexFilterValue = noSelectionText;
+
+  switch (filterValue) {
+    case "dataset-filter":
+      datasetFilterValue = filterText || noSelectionText;
+      break;
+    case "disease-filter":
+      diseaseFilterValue = filterText || noSelectionText;
+      break;
+    case "self-reported-ethnicity-filter":
+      ethnicityFilterValue = filterText || noSelectionText;
+      break;
+    case "sex-filter":
+      sexFilterValue = filterText || noSelectionText;
+      break;
+    case "no-filter":
+      break;
+    default:
+      throw new Error(`Invalid filter name: ${filterValue}`);
+  }
+
+  expect(
+    data[3].includes(`# Dataset Filter Values: ${datasetFilterValue}`)
+  ).toBeTruthy();
+  expect(
+    data[4].includes(`# Disease Filter Values: ${diseaseFilterValue}`)
+  ).toBeTruthy();
+  expect(
+    data[5].includes(
+      `# Self-Reported Ethnicity Filter Values: ${ethnicityFilterValue}`
+    )
+  ).toBeTruthy();
+  expect(
+    data[6].includes(`# Sex Filter Values: ${sexFilterValue}`)
+  ).toBeTruthy();
+};
+
+export async function downloadGeneFile(
+  page: Page,
+  tissues: string[],
+  subDirectory: string,
+  fileTypes: string[] = ["png"]
+): Promise<void> {
+  const allFileTypes = ["csv", "png", "svg"];
+  const dirPath = `${downLoadPath}/${subDirectory}`;
+
+  //wait for download file
+  const downloadPromise = page.waitForEvent("download");
+
+  //click the download icon
+  await page.getByTestId("download-button").click();
+  const CHECK = "Mui-checked";
+
+  for (const ext of allFileTypes) {
+    const checkboxId = `${ext}-checkbox`;
+    // uncheck if file type is checked but not
```

Comment:
Should we rephrase this comment for clarity?

```suggestion
    // UI has some file types selected by default. This ensures we unselect file types that don't exist in `fileTypes`
```